frontend/src/metabase/ui/components/inputs/TextareaBlurChange/TextareaBlurChange.tsx

```diff
@@ -0,0 +1,86 @@
+import type { TextareaProps } from "@mantine/core";
+import type { ChangeEvent, FocusEvent } from "react";
+import { useCallback, useLayoutEffect, useRef, useState } from "react";
+import _ from "underscore";
+
+import { useUnmountLayout } from "metabase/hooks/use-unmount-layout";
+
+import { Textarea } from "../Textarea";
+
+type TextareaRestProps = Omit<TextareaProps, "onBlur" | "ref">;
+
+export type TextareaBlurChangeProps<
+  T extends TextareaRestProps = TextareaRestProps,
+> = T & {
+  normalize?: (value?: T["value"] | undefined) => T["value"] | undefined;
+  value: T["value"] | undefined;
+  onBlurChange: (event: { target: HTMLTextAreaElement }) => void;
+};
+
+/**
+ * A wrapper around Textarea to be used with onBlurChange prop.
+ *
+ * In case you don't need it, use Textarea directly.
+ *
+ * If you're modifying this component, make the same change in TextInputBlurChange.
+ */
+export function TextareaBlurChange<T extends TextareaProps = TextareaProps>({
+  normalize = (value) => value,
+  value,
+  onBlur,
+  onBlurChange,
+  onChange,
+  ...restProps
+}: TextareaBlurChangeProps<T>) {
+  const [internalValue, setInternalValue] = useState<T["value"]>();
+  const ref = useRef<HTMLTextAreaElement>(null);
+
+  useLayoutEffect(() => setInternalValue(value), [value]);
+
+  const handleChange = useCallback(
+    (event: ChangeEvent<HTMLTextAreaElement>) => {
+      setInternalValue(event.target.value);
+
+      if (onChange) {
+        onChange(event);
+        setInternalValue(normalize(event.target.value));
+      }
+    },
+    [normalize, onChange],
+  );
+
+  const handleBlur = useCallback(
+    (event: FocusEvent<HTMLTextAreaElement>) => {
+      onBlur?.(event);
+
+      if (onBlurChange && (value || "") !== event.target.value) {
+        onBlurChange(event);
+        setInternalValue(normalize(event.target.value) ?? undefined);
+      }
+    },
+    [normalize, onBlur, onBlurChange, value],
+  );
+
+  useUnmountLayout(() => {
+    const lastPropsValue = value || "";
+    const currentValue = ref.current?.value || "";
+
+    if (ref.current && lastPropsValue !== currentValue) {
+      onBlurChange({
+        target: ref.current,
+      });
+    }
+  });
+
+  const textareaProps = _.omit(restProps, "onBlur", "onChange", "ref");
```

Comment:
!nit i don't think this is needed. `onBlur` and `onChange` are already picked off of the props and if a `ref` prop was provided it would get overwritten on line 80.

Reply:
d149e5b8a44d1ce7c73e0a35e1599e2a1c472b41